### PR TITLE
CI: Ensure author object is not null

### PR DESCRIPTION
**Pull request by aangelisc**
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by baldm0mma — APPROVED** [review]

File: .github/workflows/actions/changelog/index.js [hunk 1]
```diff
@@ -199,7 +199,13 @@ const getChangeLogItems = async (name, owner, sinceDate, to) => {
       hasLabel({ labels }, 'area/grafana/ui') ||
       hasLabel({ labels }, 'area/grafana/toolkit') ||
       hasLabel({ labels }, 'area/grafana/runtime');
-    const author = item.commits.nodes[0].commit.author.user.login;
+    
+    let author = ""
```
Comment (ifrost) [inline .github/workflows/actions/changelog/index.js:203]: You may also need to update when it's actually used in `userLink()`

File: .github/workflows/actions/changelog/index.js [hunk 2]
```diff
@@ -199,7 +199,13 @@ const getChangeLogItems = async (name, owner, sinceDate, to) => {
...
+
+    if(item.commits.nodes[0].commit.author !== null){
```
Comment (ifrost) [inline .github/workflows/actions/changelog/index.js:205]:
```suggestion
    if(item.commits.nodes[0].commit.author.user){
```

I think author is not null, it contains `{user: undefined}`

Comment (ifrost) [reply .github/workflows/actions/changelog/index.js:205]: also just doing `const author = item.commits.nodes[0].commit.author.user?.login;` should also work

File: .github/workflows/actions/changelog/index.js (hunk 1 above)
Comment (ifrost) [reply .github/workflows/actions/changelog/index.js:203]: I mean this line: https://github.com/grafana/grafana/blob/f8b092aba62115fe28116ddde663ff9909e9bf31/.github/workflows/actions/changelog/index.js#L288

We could do

```typescript
          : `${pullRequestLink(item.number)}${item.author ? ', ' + userLink(item.author) : ''}`
```

instead so the link to author is skipped

File: .github/workflows/actions/changelog/index.js [hunk 3]
```diff
@@ -160,8 +160,11 @@ const getHistory = async (name, owner, target, sinceDate) => {
       cursor,
     });
     LOG(`GraphQL: ${JSON.stringify(result)}`);
-    nodes = [...nodes, ...result.repository.object.history.nodes];
-    const { hasNextPage, endCursor } = result.repository.object.history.pageInfo;
+
+    if(result) {
```
Comment (ifrost) [inline .github/workflows/actions/changelog/index.js:164]: This may lead to skipping some results 🤔

**Pull request closed without merging**